test/functional/change_stream.test.js

```diff
@@ -2850,3 +2858,99 @@ describe('Change Streams', function() {
     });
   });
 });
+
+describe('Change Stream Resume Error Tests', function() {
+  function withChangeStream(testName, callback) {
```

Comment:
I think this might be useful as a more generalized helper `withCursor`, which we could alias to `withChangeStream` if that helps readability at all. Also, I think hiding a call to `withClient` in here might not be ideal since its sort of a side effect.

Finally, nitpick but `testName` seems to really be like `collectionName`?

Reply:
Waiting on #2362 to see how to best address this.

Reply:
I didn't ultimately find the `withDb` or `withCollection` helpers helped very much over there, but cursors are much more like clients, in that they have explicit `close` methods that sometimes need to be called.

Reply:
I added a `withCursor` helper; it's a general-purpose wrapper around `withClient`. Is this roughly what you had in mind?

```
function withCursor(getCursor, callback) {
  return withClient((client, done) => {
    getCursor(client, (err, cursor) => {
      if (err) return done(err);
      callback(cursor, () => cursor.close(done));
    });
  });
}
```

Reply:
I think of `withX` as a means of conveying the lifetime of a resource. So my thought with `withCursor` is that it would be _like_ `withClient`, but not _use_ `withClient`:
```
function withCursor(cursor, body) {
   body(cursor, (err, result) => {
     cursor.close(); // always close the cursor, this obviously would need to be async and handle errors etc
     
      callback(...);
   });
}
```

Reply:
it may not be worth the effort considering we have `this.defer`? 
```
const cursor = coll.find();
this.defer(() => cursor.close());
cursor.toArray(...);
```
could work fine, but you might forget to put the defer in between creating the cursor and iterating it